CryptoPkg/Include/Protocol/OneCrypto.h

```diff
@@ -0,0 +1,5476 @@
+/** @file
+  Implements the BaseCryptLib and TlsLib using the services of the OneCrypto
+  Protocol/PPI.
+
+  Copyright (C) Microsoft Corporation. All rights reserved.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+  UPDATED: 2025-Nov-11
+  VERSION: 1.0
+
+  ============================================================================
+  IMPORTANT: PLATFORM CODE SHOULD NOT USE THIS PROTOCOL DIRECTLY
+  ============================================================================
+
+  This protocol is an internal interface designed for communication between
+  the OneCrypto binary driver and the BaseCryptLibOnOneCrypto library. The
+  Protocol structure and its functions may change without notice.
+
+  Platform code SHOULD NOT:
+  - Locate this protocol directly using LocateProtocol/HandleProtocol
+  - Call protocol functions directly
+  - Modify or override this protocol without understanding the rules
+
+  Instead, platform code SHOULD:
+  - Use BaseCryptLibOnOneCrypto library, which provides:
+    * Version checking and compatibility validation
+    * Type safety and proper error handling
+    * A stable API that abstracts protocol implementation details
+
+  The BaseCryptLibOnOneCrypto library is the only supported way to access
+  cryptographic services and will ensure safe usage of the underlying protocol.
+  Most of these benefits will be lost without going through the library.
+
+  For more information such as updating this protocol, see Docs/ in
+  BaseCryptLibOnOneCrypto.
+
+  ============================================================================
+**/
+
+#ifndef ONE_CRYPTO_PROTOCOL_
+#define ONE_CRYPTO_PROTOCOL_
+
+#include <Library/BaseCryptLib.h>
+
+/**
+  @defgroup BaseCryptLib Base Cryptographic Library
+  @brief Provides comprehensive cryptographic services for UEFI applications.
+
+  @{
+**/
+
+// =============================================================================
+// Protocol version: 1.0
+// =============================================================================
+#define VERSION_MAJOR  1ULL
+#define VERSION_MINOR  0ULL

```

Comment:
Because C essentially has a "global namespace", these macros need to be scoped to one crypto to prevent likely name collisions.

```suggestion
#define ONE_CRYPTO_VERSION_MAJOR  1ULL
#define ONE_CRYPTO_VERSION_MINOR  0ULL
```